File: CarHomeWidget/src/main/java/com/anod/car/home/app/MusicAppChoiceActivity.java
package com.anod.car.home.app;

import com.anod.car.home.BuildConfig;
import com.anod.car.home.R;
import com.anod.car.home.model.AppsList;
import com.anod.car.home.prefs.preferences.PreferencesStorage;
import com.anod.car.home.utils.MusicUtils;

import android.app.ActivityManager;
import android.content.ComponentName;
import android.os.Bundle;
import android.view.KeyEvent;
import android.widget.CheckBox;
import android.widget.Toast;

import java.util.List;

import butterknife.ButterKnife;
import butterknife.Bind;

/**
 * @author alex
 * @date 2014-09-03
 */
public class MusicAppChoiceActivity extends MusicAppsActivity {

    @Bind(R.id.defaultApp)
    CheckBox mDefaultApp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ButterKnife.bind(this);
    }

    @Override
    protected boolean isShowTitle() {
        return false;
    }

    @Override
    protected int getFooterViewId() {
        return R.layout.music_app_choice_footer;
    }

    @Override
    protected void onEntryClick(int position, AppsList.Entry entry) {
        ComponentName musicCmp = entry.componentName;

        boolean isRunning = isMusicCmpRunning(musicCmp);

        MusicUtils.sendKeyEventComponent(
                KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE, this, musicCmp, !isRunning
        );

        if (mDefaultApp.isChecked()) {
            PreferencesStorage.saveMusicApp(this, musicCmp, true);
        }

        finish();
    }

    private boolean isMusicCmpRunning(ComponentName musicCmp) {
        ActivityManager activityManager = (ActivityManager) getSystemService(ACTIVITY_SERVICE);
        List<ActivityManager.RunningAppProcessInfo> procInfos = activityManager
                .getRunningAppProcesses();
        if (procInfos == null) {
            return false;
        }
        for (int i = 0; i < procInfos.size(); i++) {
            if (procInfos.get(i).processName.startsWith(musicCmp.getPackageName())) {
                if (BuildConfig.DEBUG) {
                    Toast.makeText(getApplicationContext(),
                            musicCmp.getPackageName() + " is running", Toast.LENGTH_LONG).show();
                }
                return true;
            }
        }
        return false;
    }

}

File: CarHomeWidget/src/main/java/com/anod/car/home/app/MusicAppChoiceActivity.kt
package com.anod.car.home.app

import android.app.ActivityManager
import android.content.ComponentName
import android.content.Context
import android.view.KeyEvent
import android.widget.CheckBox
import android.widget.Toast
import com.anod.car.home.BuildConfig
import com.anod.car.home.R
import com.anod.car.home.model.AppsList
import com.anod.car.home.utils.MusicUtils

/**
 * @author alex
 * @date 2014-09-03
 */
class MusicAppChoiceActivity : MusicAppsActivity() {

    override val footerViewId: Int
        get() = R.layout.list_footer_music_app_choice

    internal val defaultApp: CheckBox by lazy { findViewById<CheckBox>(R.id.defaultApp) }

    override fun onEntryClick(position: Int, entry: AppsList.Entry) {
        val musicCmp = entry.componentName!!

        val isRunning = isMusicCmpRunning(musicCmp)

        MusicUtils.sendKeyEventComponent(
                KeyEvent.KEYCODE_MEDIA_PLAY_PAUSE, this, musicCmp, !isRunning
        )

        if (defaultApp.isChecked) {
            val appSettings = App.provide(this).appSettings
            appSettings.musicApp = musicCmp
            appSettings.apply()
        }

        finish()
    }

    private fun isMusicCmpRunning(musicCmp: ComponentName): Boolean {
        val activityManager = getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val procInfos = activityManager
                .runningAppProcesses ?: return false
        for (i in procInfos.indices) {
            if (procInfos[i].processName.startsWith(musicCmp.packageName)) {
                if (BuildConfig.DEBUG) {
                    Toast.makeText(applicationContext,
                            musicCmp.packageName + " is running", Toast.LENGTH_LONG).show()
                }
                return true
            }
        }
        return false
    }

}
